tests/test_core.py

```diff
@@ -312,3 +312,30 @@ def test_generate_source(self):
         rs = [ts for (ti,ts) in r] 
         for (is1, is2) in zip(rs0, rs):
             compare_mstates(self, is1, is2)
+
+class TestInsertElemInModelFlame(unittest.TestCase):
+    def setUp(self):
+        testfile = os.path.join(curdir, 'lattice/test_0.lat')
+        self.testfile = make_latfile(testfile)
+
+    def test_insert_in_modelflame(self):
+        latfile = self.testfile
+        fm = ModelFlame(latfile)
+        r0,s0=fm.run(to_element=6)
+
+        new_econf = {'index':5, 'properties':{'name':'test_drift', 'type':'drift', 'L':0.05588}}
+        fm.insert_element(econf=new_econf)
+        test_econf = fm.get_element(index=5)[0]
+        self.assertEqual(test_econf['index'], new_econf['index'])
+        self.assertEqual(test_econf['properties']['name'], new_econf['properties']['name'])
+        self.assertEqual(test_econf['properties']['type'], new_econf['properties']['type'])
+        self.assertEqual(test_econf['properties']['L'], new_econf['properties']['L'])
+
+        test_econf2 = fm.get_element(index=6)[0]
+        self.assertEqual(test_econf2['index'], 6)
+        self.assertEqual(test_econf2['properties']['name'], 'LS1_CA01:BPM_D1129')
+        self.assertEqual(test_econf2['properties']['type'], 'bpm')
+
+        r1,s1 = fm.run(to_element=6)
+
+        compare_mstates(self, s0, s1)
```

Comment:
Is the inserted drift element the same as the original element with `index=6` before insertion? Or `s0` will not be equal to `s1`

Reply:
Yes. I have inserted the same length drift. At the first time, I have inserted the drift with arbitrary length and adjust the next drift length. But it causes the very small beam state difference due to the numerical error. In the `compare_mstates`, it request the all beam state value are completely the same. So I have adopted this routine.